texar/torch/modules/pretrained/t5_utils.py

```diff
@@ -0,0 +1,404 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Optional, Tuple
+import math
+import torch
+from torch.nn import functional as F
+from torch import nn
+
+from texar.torch.core import layers
+from texar.torch.module_base import ModuleBase
+from texar.torch.modules.encoders.multihead_attention import LayerCache
+
+
+from texar.torch.utils.types import MaybeList
+
+__all__ = [
+    "T5LayerNorm",
+    "MultiheadRPRAttention"
+]
+
+
+class T5LayerNorm(nn.Module):
+    r""" Custom LayerNorm for T5 with no mean subtraction and no bias.
+    """
+
+    def __init__(self, input_size: int, eps: float = 1e-5):
+        super().__init__()
+
+        self.w = nn.Parameter(torch.ones(input_size))
+        self.eps = eps
+
+    def forward(self,  # type: ignore
+                x: torch.Tensor):
+        x = x / torch.sqrt(x.pow(2).mean(-1, keepdim=True) + self.eps)
+        return self.w * x
+
+
+class MultiheadRPRAttention(ModuleBase):
```

Comment:
I'm not very familiar with this paper, is the relative attention here very different from the one used in Transformer-XL (and subsequently XLNet)? It's fine to have a separate module (much less error-prone, I assume), but we might be able to merge them into one class some time in the future.